### DISCUSSION TITLE
Case studies

### DISCUSSION THREAD

**Discussion by ChristineStawitz-NOAA** [NOAA-FIMS/FIMS#530] [Q&A]
A place to ask questions about FIMS case studies

**Comment by ChristineStawitz-NOAA** [comment]
Should we do this here or in the Google space?

**Comment by cmlegault** [comment]
I just uploaded asap2fims_v2.R [here](https://drive.google.com/drive/folders/1WD8eilQORw1rq1aEQr7LFWYlbzSmWwqP?usp=drive_link). It now runs with the latest and greatest version of FIMS.

**Comment by msupernaw** [comment]
Interesting results! NAA looks pretty good, however FIMS seems to estimate
lower SSB values. Did FIMS meet the convergence criteria?

On Thu, Nov 30, 2023 at 2:39 PM Chris Legault ***@***.***>
wrote:

> I just uploaded asap2fims_v2.R here
> <https://drive.google.com/drive/folders/1WD8eilQORw1rq1aEQr7LFWYlbzSmWwqP?usp=drive_link>.
> It now runs with the latest and greatest version of FIMS.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/NOAA-FIMS/discussions/530#discussioncomment-7721692>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABFUSEBJHMJZTQ6HUB5MVPTYHDOFTAVCNFSM6AAAAAA76SKKDSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TOMRRGY4TE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Matthew Supernaw
*Scientific Software Developer*
*National Oceanic and Atmospheric Administration*
*Office Of Science and Technology*
*NOAA Fisheries | *U.S. Department of Commerce
Phone 248 - 396 - 7797

**Comment by cmlegault** [comment]
I used the additional Newton step trick from Cole to get really good convergence in FIMS. The SSB values are due to an undocumented feature in the FIMS SSB calculations that applies a 0.5 multiplier (see Issue #521).

**Comment by Cole-Monnahan-NOAA** [comment]
I'm getting an error when running this line

`fishing_fleet$log_obs_error$value <- log(sqrt(log(as.numeric(mean(rdat$control.parms$catch.tot.cv[,1], na.rm=TRUE)^2) + 1))) `

> Error: Not compatible with requested type: [type=list; target=double].
In addition: Warning message:
In fishing_fleet$log_obs_error$value <- -2 : Coercing LHS to a list

Should I go back to a previous commit?

**Reply by iantaylor-NOAA** [reply]
If you're using current main branch, then I think the `log_obs_error` needs to be a vector based on this change: https://github.com/NOAA-FIMS/FIMS/pull/532/files.

**Comment by Cole-Monnahan-NOAA** [comment]
I tried that and got the same error so I went back to the commit referenced in the code and now it seems to run.

**Reply by ChristineStawitz-NOAA** [reply]
Thanks @iantaylor-NOAA you are correct. In addition to changing the input, the names of these two values need to be changed wherever they are called to reflect that it's a vector and not a scalar quantity anymore:

```
fishing_fleet$log_obs_error <- rep(log(0.01^2 + 1))), nyears)
fishing_fleet$estimate_obs_error <- FALSE
```

See here for full syntax: https://noaa-fims.github.io/FIMS/articles/fims-demo.html

**Comment by Cole-Monnahan-NOAA** [comment]
I got pretty far on bridging my case study but am finding some strange behavior with the initial NAA and plus group calculations. Perhaps this could be a seaside chat topic?

**Reply by ChristineStawitz-NOAA** [reply]
Thanks! Can you make the seaside chat tomorrow to discuss?

**Reply by Cole-Monnahan-NOAA** [reply]
Yes I hope to be there unless I'm still feeling sick. I have some really informal stuff prepared that we can walk through.

**Comment by kellijohnson-NOAA** [comment]
I am going to close this Discussion because it is quite stale and if there are problems with the case studies we can open more specific Discussion questions or post Issues. Feel free to open it back up if needed.
